takhojin/services/userService.js

```diff
@@ -1,14 +1,47 @@
-const bcrypt = require("bcrypt");
-
 const userDao = require("../models/userDao");
 const { pwValidation } = require("../utils/validation-check");
 
+const bcrypt = require("bcrypt");
+
+const jwt = require("jsonwebtoken");
+
+const secretKey = process.env.secretKey;
+const salt = 13;
+
+const makeHash = async (password, salt) => {
+  return await bcrypt.hash(password, salt);
+};
+
+const checkHash = async (password, hashedpassword) => {
+  return await bcrypt.compare(password, hashedpassword);
+};
+
 const signUp = async (email, password, description, profileImg) => {
   await pwValidation(password);
-  const salt = 13;
-  const hashPassword = await bcrypt.hash(password, salt);
 
-  return userDao.createUser(email, hashPassword, description, profileImg);
+  const hashPassword = await makeHash(password, salt);
+
+  return await userDao.createUser(email, hashPassword, description, profileImg);
+};
+
+const login = async (email, password) => {
+  await pwValidation(password);
+
+  const hashedPasswordObj = await userDao.login(email);
+  const hashedPassword = hashedPasswordObj[0].password;
+  const hashedPasswordId = hashedPasswordObj[0].id;
+
+  const checkResult = await checkHash(password, hashedPassword);
+
+  if (checkResult === true) {
+    const payLoad = { userId: hashedPasswordId };
+    return await jwt.sign(payLoad, secretKey);
+  } else {
+    const err = new Error("INVALID_USER");
+    console.log(err);
+    err.statuscode = 400;
+    throw err;
+  }
```

Comment:
```suggestion
  const [user] = await userDao.getUserByEmail(email);
  
  if(!user) throw new Error('INVALID_EMAIL_OR_PASSWORD')

  const checkResult = await checkHash(password, user.password);

  if (!checkResult) throw new Error('INVALID_EMAIL_OR_PASSWORD');
    
  return jwt.sign({userId : user.id}, process.env.JWT_SECRET);
```
suggestion에 남겨드린 코드 비교해시면 좋을 것 같습니다.
1. 로그인 시도의 email이 잘못된 경우, password가 잘못된 경우 모두 핸들링을 해야합니다.
2. payload와 secret key 부분의 값이 복잡한 객체가 아니라면 직접 작성해주는 것이 보다 깔끔합니다.
3. 성공의 case를 if문으로 따지는 것보다 각 예외 case들을 if문을 통해 검증한뒤에 성공을 return하는 로직으로 작성되어야 제대로된 예외처리가 가능합니다!